diff -Nur pesign-113/src/pesigcheck.c pesign-113-new/src/pesigcheck.c
--- pesign-113/src/pesigcheck.c	2019-05-11 02:53:51.000000000 +0800
+++ pesign-113-new/src/pesigcheck.c	2021-07-30 11:25:25.000000000 +0800
@@ -318,7 +318,6 @@
 			reason->type = SIGNATURE;
 			reason->sig.data = data;
 			reason->sig.len = datalen;
-			reason->type = siBuffer;
 			nreason += 1;
 			is_invalid = true;
 		}
@@ -330,7 +329,6 @@
 			reason->type = SIGNATURE;
 			reason->sig.data = data;
 			reason->sig.len = datalen;
-			reason->type = siBuffer;
 			nreason += 1;
 			has_valid_cert = true;
 		}
